### ISSUE TITLE
glyphs2ufo has incorrect GDEF

### ISSUE THREAD

**Issue by kontur** [googlefonts/glyphsLib#620]
Running:

```
glyphs2ufo --generate-GDEF Handjet.glyphs
```

on [this file](https://github.com/rosettatype/Handjet/blob/1220be6b70a319c46783a0e76cd387e55d66ee42/sources/Handjet.glyphs) will generate a GDEF that is missing several mark definitions, for example `dieresisbelow` (U+0324).

The glyph has zero width and is defined a non-spacing mark in the GlyphsData.xml I am using:
<img width="943" alt="Screenshot 2020-09-07 at 9 12 28" src="https://user-images.githubusercontent.com/1893577/92354094-522e4900-f0ea-11ea-859c-e1d2c70596ed.png">

This issue became apparent when reviewing [Handjet with Fontbakery](https://github.com/rosettatype/Handjet/issues/8).

**Comment by madig** [comment]
glyphsLib is hardcoded to use the cut-down GlyphData.xml file supplied at https://github.com/schriftgestalt/GlyphsInfo. You can either insert a step in your piepline to change the custom names back or send a PR to make it possible to use additional databases, but I currently have no resources to look at any of it.

**Comment by schriftgestalt** [comment]
The name of the glyph with unicode 0324 should be `dieresisbelowcomb`. Your screenshot shows the name as `dieresisbelow`. What version of Glyphs do you have? Do you have a custom GlyphData.xml? If so, you should update it. Only include the glyph that you are actually changing in your custom file and don’t copy the full file. 

But maybe the info lookup in glyphsLib could be improved to also look for the unicode?

**Comment by madig** [comment]
Using the glyph data is enough of a headache as is, I'd only want to implement what Glyphs does.

**Comment by schriftgestalt** [comment]
Glyphs is falling back to the unicode to find the info.
<img width="305" alt="Screenshot" src="https://user-images.githubusercontent.com/174660/92380653-db08ad00-f109-11ea-9878-8146e920c90f.png">
This is with default GlyphData that doesn't mention `dieresisbelow`

**Comment by kontur** [comment]
Should have pointed out that I am using a custom GlyphsData.xml — doesn't the screenshot indicate that despite "our" name being `diesresisbelow` (without the "comb") the groups/categories are correct?

I was under the impression that `glyphsLib` works with the unicode values? Are you saying that running `glyphsLib` with any source that follows a custom GlyphsData.xml it is at best happenstance what GDEF groups `glyphsLib` will allocate? :/

**Comment by madig** [comment]
> I was under the impression that glyphsLib works with the unicode values?

I can't remember what it works with, but it's convoluted because that's what Glyphs does, see https://github.com/googlefonts/glyphsLib/blob/master/Lib/glyphsLib/glyphdata.py.

> Are you saying that running glyphsLib with any source that follows a custom GlyphsData.xml it is at best happenstance what GDEF groups glyphsLib will allocate? :/

Yes.

**Comment by kontur** [comment]
Oh! Major bummer, and I was not aware of that — but of course it makes sense, the .glyphs source has no record of its creation GlyphsApp environment aside the unicode point and why should this library be able to access my local config files from some app. 🤦 

@madig / @schriftgestalt would you take a look at the above commit? I got it to work™ but I am sure the parameter passing and relying on a global is not the most elegant way to achieve this.

E.g. this should work:

```
glyphs2ufo --glyphdata=/path/to/Glyphs/Info/GlyphData.xml Source.glyphs
```

Since the GlyphsData is loaded by way of first access I wasn't quite sure how to pass such a CLI parameter down the line. I ended up with `GLYPHSDATA_PATHS` in `builder/builders.py`, so the instantiation of `UFOBuilder` can overwrite that and `glyphsdata.py` can import that when eventually called, or load the included files form the package if empty.

**Comment by kontur** [comment]
Thanks @schriftgestalt for some obvious catches. Can I somehow show a diff to [my branch](https://github.com/kontur/glyphsLib/tree/glyphdata) without making a PR so it's easier to review the aggregate changes of both commits?

**Comment by davelab6** [comment]
Yes, you an use the compare feature: https://github.com/googlefonts/glyphsLib/compare/master...kontur:glyphdata

**Comment by khaledhosny** [comment]
Was a PR made for this, if not I think that is the proper way to get the code reviewed. You can [mark the PR draft](https://github.blog/2019-02-14-introducing-draft-pull-requests/) if you think it is not ready yet.

**Comment by yanone** [comment]
I have a similar issue. Many glyphs in an Indic font I'm working on have non-standard glyph names but the correct unicodes. fontmake doesn’t pick them into the mark category in GDEF by its name, tho it could by its unicode. Is that not something that should be implemented? Georg mentioned that Glyphs.app uses a unicode fallback, and the font does generate correctly in Glyphs.app despite the non-standard glyph names.

**Comment by khaledhosny** [comment]
Fixed with #668
